cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -216,9 +228,19 @@ func (g *gitHubRepository) getVersions() ([]string, error) {
 
 	// get all the releases
 	// NB. currently Github API does not support result ordering, so it not possible to limit results
-	releases, _, err := client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, nil)
-	if err != nil {
-		return nil, g.handleGithubErr(err, "failed to get the list of releases")
+	var releases []*github.RepositoryRelease
+	var retryError error
+	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
+		var listReleasesErr error
```

Comment:
```suggestion
		var listReleasesErr error
		retryError = nil
```
We should reset the error inside the block here. If not we will return an error even if a first attempt failed but a subsequent attempt succeeded.

Same comment at the other sections.